src/hssm/utils.py

```diff
@@ -146,6 +153,203 @@ def _get_alias_dict(
     return alias_dict
 
 
+def _compute_log_likelihood(
+    model: bmb.Model,
+    idata: az.InferenceData,
+    data: pd.DataFrame | None,
+    inplace: bool = True,
+) -> az.InferenceData | None:
+    """Compute the model's log-likelihood.
+
+    Parameters
+    ----------
+    idata : InferenceData
+        The `InferenceData` instance returned by `.fit()`.
+    data : pandas.DataFrame or None
+        An optional data frame with values for the predictors and the response on which
+        the model's log-likelihood function is evaluated.
+        If omitted, the original dataset is used.
+    inplace : bool
+        If True` it will modify `idata` in-place. Otherwise, it will return a copy of
+        `idata` with the `log_likelihood` group added.
+
+    Returns
+    -------
+    InferenceData or None
+    """
+    # These are not formal parameters because it does not make sense to...
+    #   1. compute the log-likelihood omitting
+    #      the group-specific components of the model.
+    #   2. compute the log-likelihood on unseen groups.
+    include_group_specific = True
+    sample_new_groups = False
+
+    # Get the aliased response name
+    response_aliased_name = get_aliased_name(model.response_component.term)
+
+    if not inplace:
+        idata = deepcopy(idata)
```

Comment:
`deepcopy` is very expensive especially since `idata` can be pretty big. A shallow copy is sufficient in this context

Reply:
in the end decided to leave this after some back and forth.